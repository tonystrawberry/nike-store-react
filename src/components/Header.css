header{
  height: 80px;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  width: 100%;
  padding-left: 8px;
  padding-right: 8px;
  box-shadow: inset 0 -1px 0 0 #e5e5e5;
  background: #fff;
  line-height: 80px;
  z-index: 100;
  display: flex;
}

header img{
  width: 80px;
  vertical-align: middle;
  margin-right: 20px;
  margin-left: 20px;
}

.admin{
  margin-left: auto;
}

.admin img{
  width: 20px;
}