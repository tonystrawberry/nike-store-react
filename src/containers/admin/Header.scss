header {
  height: 80px;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  width: 100%;
  box-shadow: inset 0 -1px 0 0 #e5e5e5;
  background: #fff;
  z-index: 100;
  display: flex;

  > a {
    display: flex;
    align-items: center;
  }

  img {
    vertical-align: middle;
    margin-right: 20px;
    margin-left: 20px;
  }

  .logo {
    width: 80px;
  }
}

.right {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-left: auto;
  padding: 0 5px;

  .button {
    margin: 0 10px;
  }
}

.menu{
  position: relative;
  img {
    width: 20px;

    &:hover{
      cursor: pointer;
    }
  }
}

.menu__dropdown {
  display: none;
  position: absolute;
  top: 100%;
  right: 0;
  margin: 10px;
  border: 1px solid #e5e5e5;
  width: 300px;

  &.active {
    display: block;
  }

  ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
    width: 100%;
    li {
      text-align: center;

      &:hover {
        cursor: pointer;
        background-color: #fbfbfb;
        color: gray;
      }
      
      &:not(:last-child) {
        box-shadow: inset 0 -1px 0 0 #e5e5e5;
      }

      a {
        padding: 10px;
        width: 100%;
        height: 100%;
        display: block;
        text-align: center;
        color: #333;
        text-decoration: none;
      }
    }
  }

}

.menu__dropdown-mobile {
  display: none;
  height: 100vh;
  width: 100%;
  background: rgba(0, 0, 0, 0.7);
  position: absolute;
  height: 100vh;
  width: 100%;
  margin: 0;
  padding: 0;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  list-style-type: none;
  font-size: 25px;

  > div {
    margin-top: 10px;

    &.close {
      margin-top: 50px;
    }
  }

  a {
    color: #fff;
    text-decoration: none;
  }

}

/* Responsive */
@media (max-width: 1070px) {

  .menu{
    position: static;
  }

  .menu__dropdown-mobile {
    &.active {
      display: flex;
    }

  }

  .menu__dropdown {
    &.active {
      display: none;
    }
  }
}