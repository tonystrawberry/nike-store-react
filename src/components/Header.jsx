import React, { PureComponent } from 'react';
import logo from '../assets/nike_logo.svg';
import './Header.css';

class Header extends PureComponent {
  render() {
    return (
      <header>
        <a href="/"><img className="logo" src={logo} alt="Nike" /></a>
      </header>
    );
  }
};

export default Header;