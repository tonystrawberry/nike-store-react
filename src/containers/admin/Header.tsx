import React, { Component } from 'react';
import logo from '../../assets/nike_logo.svg';
import menu from '../../assets/menu.svg';

import cart from '../../assets/cart.svg';
import './Header.scss';
import { Link } from 'react-router-dom';
import { AdminUser } from '../../types';
import close from '../../assets/close.svg';

interface IHeaderProps {
  connected: boolean,
  user: AdminUser | null
}

interface IHeaderState {
  menuActive: boolean
}

class Header extends Component<IHeaderProps, IHeaderState> {
  constructor(props: IHeaderProps){
    super(props)

    this.state = {
      menuActive: false
    }
  }

  render(){
    return (
      <header>
        <Link to="/"><img className="logo" src={logo} alt="Nike" /></Link>
        {
          this.props.connected ? 
            <React.Fragment>
              <div className="menu right">
                <Link className="cart" to="/cart"><img width="20" src={cart} alt="Cart" /></Link>
                {/* <Link className="menu right" to="/admin"><img width="20" src={menu} alt="Menu" /></Link> */}
                <img width="20" src={menu} alt="Menu" onClick={() => { this.setState({menuActive: !this.state.menuActive}) }} />
              </div>
              <div className={`menu__dropdown-mobile ${this.state.menuActive ? 'active' : ''}`}>
                <div><Link className="" to="/admin/products" onClick={() => { this.setState({ menuActive: false }) }}>ACCOUNT</Link></div>
                <div className="close"><img width="40px" src={close} onClick={() => this.setState({menuActive: !this.state.menuActive})} alt="Close"></img></div>
              </div>

              <div className={`menu__dropdown ${this.state.menuActive ? 'active' : ''}`}>
                <ul>
                  <li><Link className="menu right" to="/admin/products" onClick={() => { this.setState({ menuActive: false }) }}>ACCOUNT</Link></li>
                </ul>
              </div>
            </React.Fragment>
          : 
            <React.Fragment>
              <div className="right">
                <Link className="cart" to="/cart"><img width="20" src={cart} alt="Cart" /></Link>
                <Link className="login" to="/login"><button className="button">Sign in</button></Link>
              </div>
            </React.Fragment>
        }
      </header>
    );
  }
};

export default Header;