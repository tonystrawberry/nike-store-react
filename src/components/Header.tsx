import React from 'react';
import logo from '../assets/nike_logo.svg';
import admin from '../assets/admin.svg';
import './Header.scss';
import { Link } from 'react-router-dom';
import { AdminUser } from '../types';

interface IHeaderProps {
  connected: boolean,
  user: AdminUser | null
}

const Header = (props: IHeaderProps) => {
  return (
    <header>
      <Link to="/"><img className="logo" src={logo} alt="Nike" /></Link>
      {
        props.connected ? 
          <React.Fragment>
            <div className="right">
              <div>Welcome {props.user != null ? props.user.username : ''}</div>
              <Link className="admin right" to="/admin"><img width="20" src={admin} alt="Admin" /></Link>
            </div>
          </React.Fragment>
        : 
          <React.Fragment>
            <div className="right">
              <Link className="login" to="/login"><button className="button">Sign in</button></Link>
            </div>
          </React.Fragment>
      }
    </header>
  );
};

export default Header;