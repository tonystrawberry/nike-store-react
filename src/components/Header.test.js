import { shallow } from 'enzyme';
import React from 'react';
import Header from './Header';
import toJson from 'enzyme-to-json';

it ('expect to render Header component', () => {
  expect(toJson(shallow(<Header />))).toMatchSnapshot();
})
