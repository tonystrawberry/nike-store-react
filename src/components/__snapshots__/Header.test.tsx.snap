// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Header component 1`] = `
<header>
  <a
    href="/"
  >
    <img
      alt="Nike"
      className="logo"
      src="nike_logo.svg"
    />
  </a>
</header>
`;
